src/clients/BundleDataClient/utils/PoolRebalanceUtils.ts

```diff
@@ -202,6 +202,9 @@ export function updateRunningBalanceForDeposit(
     deposit.originChainId,
     mainnetBundleEndBlock
   );
+  if (!l1TokenCounterpart) {
```

Comment:
I think we might want to assert here instead of return early, since this is an instance where it's definitely a bug if this is undefined, so we wouldn't want to silently fail.